testing/typescript/data/custom_store.ts

```diff
@@ -46,3 +49,22 @@ export function loadAcceptsPromiseOfObject() {
     load: () => new Promise<object>(undefined),
   });
 }
+
+export function promiseResolveAcceptsMultipleArguments() {
+  const value = {
+    id: 1,
+    text: 'a',
+  };
+  const store = new CustomStore({
+    load: () => new Promise((resolve) => {
+      resolve((v1, v2: { totalCount: number; groupCount: number }) => {});
+    }),
+  });
+  store.insert(value).then((data, key: number) => ({ data, key }));
+  store.update(1, value).then((data, key: number) => ({ data, key }));
+
+  const dataSource = new DataSource(store);
+  dataSource.load().then(
+    (data, extra: { totalCount: number; groupCount: number }) => ({ data, extra }),
+  );
```

Comment:
A test should be as short as possible.

I think it can be reduced to this:

```ts
/* eslint-disable @typescript-eslint/no-unnecessary-type-arguments */
/* eslint-disable @typescript-eslint/init-declarations */
/* eslint-disable @typescript-eslint/no-explicit-any */
...

const ANY = undefined as any;
...

export function promiseResolveAcceptsMultipleArguments() {
  const callback: (a: any, b: any) => void = ANY;

  const store: CustomStore<any, any> = ANY;
  store.load().then(callback);
  store.load(ANY).then(callback);
  store.insert(ANY).then(callback);
  store.update(ANY, ANY).then(callback);
}

```

Reply:
It seems that same compile cases should be checked for stores of other kinds. DataSource's cases should be moved to a separate file as well (or rename `custom_store.ts` to something more generic).

Reply:
done